### PR TITLE
yarp: add python bindings

### PR DESCRIPTION
**Pull request by kt10aan**
Added ''--with-python'' option.
Python bindings were broken in YARP. Now they are fixed and the option is added in the formula.

### CODE REVIEW

File: yarp.rb [hunk 1]
```diff
@@ -29,7 +30,9 @@ class Yarp < Formula
   depends_on "homebrew/science/opencv" => :optional
   depends_on "qt5" => :optional
   depends_on "lua" => :optional
+  depends_on "python" => :optional
   depends_on "swig" if build.with? "lua"
```
Comment (DomT4) [inline yarp.rb]: Let's block this:

```ruby
if build.with?("lua") || build.with?("python")
  depends_on "swig"
end
```

**Comment by kt10aan** [comment]
Updated :)

**Comment by DomT4** [comment]
Some whitespacing is present apparently:

```
Error: 2 problems in 1 formula
homebrew/x11/yarp:
 * 34: Trailing whitespace was found
 * 61: Trailing whitespace was found
```

**Comment by kt10aan** [comment]
Ooops! Fixed :)

**Comment by kt10aan** [comment]
Regarding the above fail: The ``--with-python`` option works only with the ``--HEAD`` option.
What is the recommended way to add the option only if the formula is being built from ``--HEAD``

**Comment by DomT4** [comment]
Tuck everything inside a `head do` block. You'll need to revert the `swig` change as well. Something like:

```ruby
head do
  url "blah"

  option "with-python", "Install Python Bindings"

  depends_on "python" => :optional
  depends_on "swig" if build.with? "python"
end
```

And then in `def install`:

```ruby
if build.head? && build.with?("python")
  blah
  blah
end
```

Should work fine.

**Comment by kt10aan** [comment]
Thanks for the help! I updated the formula according to your suggestions. The checks failed but I think it is the browbot acting up.

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by dunn** [comment]
Merged in e5698cd8942b8f47440dec86107b920e357046d0, thanks!

Could you ask upstream to tweak their linking before the next stable release?
```
Warning: python modules have explicit framework links
These python extension modules were linked directly to a Python
framework binary. They should be linked with -undefined dynamic_lookup
instead of -lpython or -framework Python.
  /usr/local/Cellar/yarp/HEAD/lib/python2.7/site-packages/_yarp.so
```

**Pull request closed without merging**